src/parser/SparqlParser.cpp

```diff
@@ -293,8 +290,10 @@ void SparqlParser::parseWhere(ParsedQuery* query,
       GraphPatternOperation::Bind bind{parseExpressionWithAntlr()};
       _lexer.expect("as");
       _lexer.expect(SparqlToken::Type::VARIABLE);
-      query->selectClause()._varsOrAsterisk.addVariableFromQueryBody(
-          _lexer.current().raw);
+      if (!query->addVariableFromSubQueryBody(query, _lexer.current().raw)) {
+        LOG(DEBUG) << "Cannot add variable " + _lexer.current().raw +
+                          " because there is not a Select Clause in the query";
+      }
```

Comment:
There should be no log here (and also not in all the other cases). This is not an unusual / unexpected behavior.
(With my suggestion from above, the function is not `[[nodiscard]]` anymore, so you can ignore the return value.